src/components/tree-views-container.js

```diff
@@ -47,6 +86,196 @@ const TreeViewsContainer = () => {
 
     const { enqueueSnackbar } = useSnackbar();
 
+    /* Dialogs states */
+    const [anchorEl, setAnchorEl] = React.useState(null);
+    const [openAddNewStudyDialog, setOpenAddNewStudyDialog] = React.useState(
+        false
+    );
+    const [
+        openCreateNewDirectoryDialog,
+        setOpenCreateNewDirectoryDialog,
+    ] = React.useState(false);
+    const [
+        openDeleteDirectoryDialog,
+        setOpenDeleteDirectoryDialog,
+    ] = React.useState(false);
+    const [
+        openCreateRootDirectoryDialog,
+        setOpenCreateRootDirectoryDialog,
+    ] = React.useState(false);
+    const [
+        openRenameDirectoryDialog,
+        setOpenRenameDirectoryDialog,
+    ] = React.useState(false);
+    const [
+        openAccessRightsDirectoryDialog,
+        setOpenAccessRightsDirectoryDialog,
+    ] = React.useState(false);
+
+    const [accessRightsError, setAccessRightsError] = React.useState('');
+    const [deleteError, setDeleteError] = React.useState('');
+    const [renameError, setRenameError] = React.useState('');
+
+    /* Menu states */
+    const [mousePosition, setMousePosition] = React.useState(
+        initialMousePosition
+    );
+
+    const [
+        showCreateStudyMenuItem,
+        setShowCreateStudyMenuItem,
+    ] = React.useState(true);
+    const [
+        showCreateNewDirectoryMenuItem,
+        setShowCreateNewDirectoryMenuItem,
+    ] = React.useState(true);
+
+    /* User interactions */
+    const onContextMenu = useCallback(
+        (e, nodeId) => {
+            setMousePosition({
+                mouseX: e.clientX + 16,
+                mouseY: e.clientY - 4,
+            });
+            /* open Contextual Menu in empty space */
+            if (!nodeId) {
+                setShowCreateStudyMenuItem(false);
+                setShowCreateNewDirectoryMenuItem(false);
+            } /* open Contextual Menu in a TreeViewItem */ else {
+                setShowCreateStudyMenuItem(true);
+                setShowCreateNewDirectoryMenuItem(true);
+            }
+
+            if (nodeId !== selectedDirectoryRef.current) {
+                dispatch(setSelectedDirectory(nodeId));
```

Comment:
i think this is already handled , isn't it?

Reply:
ok but I'm not sure we should allow to call `dispatch(setSelectedDirectory(null));` when the contextual menu is opened from the empty space, because it changes the content and breadcrumbs display and it should be made only if we create a new root directory effectively.
I propose to amend the code with this test only to test if nodeId is null or not:
`if (nodeId) `

Reply:
ok resolved by calling handleselect when rightclick on a treeitem before openning the contextual menu